### ISSUE TITLE
illumos toolchain builds

### ISSUE THREAD

**Issue by jclulow** [rust-lang/compiler-team#279]
# TL;DR

Now that the [illumos](https://illumos.org) target has been integrated into Rust (see rust-lang/rust#55553 and rust-lang/rust#71145) we would like to have the official project CI produce `x86_64-unknown-illumos` toolchain binaries that can be installed with `rustup`.  We are keen to build Rust-based software on illumos systems, and indeed even to allow Rust as an implementation language alongside C for core illumos programs, libraries, and kernel modules.

# Links and Details

[illumos](https://illumos.org) is an open source UNIX operating system, with several actively maintained distributions such as [OmniOS](https://omniosce.org/) and [SmartOS](https://www.joyent.com/smartos).  Following some work by Rust community members, staff at Joyent, and most recently @pfmooney and myself, the toolchain and many crates are working quite well on our platform.

The Rust ecosystem has done a great job in producing documentation and tools, and in particular with easy toolchain management through `rustup`.  We would like to get illumos toolchains to be built along with all of the other nightly (and eventually beta/stable) toolchains that are built today, so that users on illumos can also use `rustup` to install official binaries.  Through some initial testing, we've been able to produce a functional prototype `rustup` that works on various illumos distributions today, albeit from one of our own webservers and with only the binaries we have built by hand.

In addition to binary availability through `rustup` we are keen to be active maintainers of our port, helping to ensure that any platform-specific issues are caught during CI and that we can help with any required fixes.  To that end, @pfmooney and I (@jclulow) would agree to sign up to whatever GitHub group and/or to be notified or consulted on any issues that appear illumos-specific.  I'm a member of the illumos core team, and Patrick has been working on illumos for quite some time, so we should be well-placed to help.

As part of our attempt to add an illumos host built to the CI environment, we initially created a new `dist-x86_64-illumos` image, in a similar vein to the existing `dist-x86_64-freebsd` and `-netbsd` builder images.  Unfortunately it doesn't seem like there is capacity in the CI environment for an entirely new image, so as requested we have instead folded the build into the existing `dist-various-2` builder image.  That work is ongoing in rust-lang/rust#71272, which is what spurred the creation of this MCP.

# Mentors or Reviewers

None as yet.  Would welcome guidance!  It seems like we at least need some folks from the infrastructure/CI side of things as well as whatever folks from the toolchain side.

**Comment by davidtwco** [comment]
@rustbot second

Personally, I'm interested in seeing better illumos support in Rust, it's a fascinating target. rust-lang/rust#71272 looks to be progressing and this MCP lists folks who would be willing to help review and maintain the target.

**Comment by nikomatsakis** [comment]
It's not clear at the moment what protocol to use for approving things for distribution. But in our discussion on https://github.com/rust-lang/compiler-team/issues/312 we decided to start with a full check-off, since enabling rustup distribution is something that is hard (though not impossible, for sure) to walk back.

Therefore, I would like to propose to @rust-lang/compiler that we elevate illumos to a rustup-distributed target. I've discussed with @jclulow and gotten a preliminary list of names of folks that would help with maintenance. 

@rfcbot fcp merge

**Comment by nikomatsakis** [comment]
PS sorry @jclulow for the procedural hiccups, bear with us :)

**Comment by nikomatsakis** [comment]
It's not clear at the moment what protocol to use for approving things for distribution. But in our discussion on #312 we decided to start with a full check-off, since enabling rustup distribution is something that is hard (though not impossible, for sure) to walk back.

Therefore, I would like to propose to @rust-lang/compiler that we elevate illumos to a rustup-distributed target. I've discussed with @jclulow and gotten a preliminary list of names of folks that would help with maintenance.

@rfcbot fcp merge

**Comment by jclulow** [comment]
> PS sorry @jclulow for the procedural hiccups, bear with us :)

No worries!  Thanks for taking the time!
